### PR TITLE
Implementing getAll() for GoogleTasksDataAccessObject

### PR DESCRIPTION
**Pull request by nnzahin**
Implementing getAll() for search use case

### CODE REVIEW

**Review by thatlittlegit — CHANGES_REQUESTED** [review]
Thanks for working on it! Main issue is that the `idsToUUIDs` probably won't work, see the comment I left. Otherwise, I have a handful of other minor issues with it, as mentioned.

Also, _if_ it wouldn't be too much hassle, would you mind writing a few test cases? I think I have some in TaskAccount which could be used as inspiration; the idea is to use a constant APIProvider and see how it handles the responses.

Again, though, thanks for working on it! We can't be that far from having search working now :D

File: src/main/java/somethingrandom/dataaccess/google/tasks/TaskList.java [hunk 1]
```diff
@@ -34,4 +37,22 @@ public void add(Item item) throws DataAccessException, IOException {
         request.put("notes" , item.toString());
         provider.request(new APIRequestBody.JSONBody("POST", request), "https://tasks.googleapis.com/tasks/v1/lists/" + identifier + "/tasks");
     }
+
+    public Collection<Item> getAll() throws AuthenticationException, IOException {
+        try {
+            JSONObject response = provider.request(new APIRequestBody.JSONBody("GET", new JSONObject()), "https://tasks.googleapis.com/tasks/v1/lists/" + identifier + "/tasks");
+            JSONObject items = response.getJSONObject("items");
+            Collection<Item> allItems = new ArrayList<>();
+            JsonItemFactory factory = new JsonItemFactory();
```
[the remark this answers is not in the document]
Comment (nnzahin) [reply src/main/java/somethingrandom/dataaccess/google/tasks/TaskList.java:46]: Static should be ok

File: src/main/java/somethingrandom/dataaccess/google/tasks/TaskList.java [hunk 2]
```diff
@@ -34,4 +38,21 @@ public void add(Item item) throws DataAccessException, IOException {
         request.put("notes" , item.toString());
         provider.request(new APIRequestBody.JSONBody("POST", request), "https://tasks.googleapis.com/tasks/v1/lists/" + identifier + "/tasks");
     }
+
+    public Collection<Item> getAll() throws AuthenticationException, IOException {
+        JSONObject response = provider.request(new APIRequestBody.JSONBody("GET", new JSONObject()), "https://tasks.googleapis.com/tasks/v1/lists/" + identifier + "/tasks");
+        JSONObject items = response.getJSONObject("items");
+        Collection<Item> allItems = new ArrayList<>();
+        for (Object item: response.getJSONArray("items")) {
+            JSONObject jsonItem = new JSONObject(item);
```
Comment (thatlittlegit) [inline src/main/java/somethingrandom/dataaccess/google/tasks/TaskList.java:47]: Should've pointed this out the first time: I don't think this is right. The [documentation](https://stleary.github.io/JSON-java/org/json/JSONObject.html) says:

> [JSONObject](https://stleary.github.io/JSON-java/org/json/JSONObject.html#JSONObject-java.lang.Object-)([Object](https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true) bean)
Construct a JSONObject from an Object using bean getters.

But JSONObject isn't a bean. Probably `(JSONObject) item` would work?

**Review by thatlittlegit — APPROVED** [review]
Some tests would be nice in future, but looks good!

File: src/main/java/somethingrandom/dataaccess/google/tasks/GoogleTasksDataAccessObject.java [hunk 3]
```diff
@@ -32,4 +34,13 @@ public void save(Item item) throws DataAccessException {
             throw new DataAccessException(e);
         }
     }
+
+    @Override
+    public Collection<Item> getAll() {
```
Comment (fahimfa3) [inline src/main/java/somethingrandom/dataaccess/google/tasks/GoogleTasksDataAccessObject.java]: Can you rename this to getAllItems()? That's the name in the Search Data Access Interface

**Pull request merged**